File: android/lib/src/test/java/com/deanalvero/answers/leetcode/easy/DefangingAnIPAddressTest.java
package com.deanalvero.answers.leetcode.easy;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class DefangingAnIPAddressTest {

    private DefangingAnIPAddress testee;

    @Before
    public void setUp() {
        testee = new DefangingAnIPAddress();
    }

    @Test
    public void defangIPaddr_inputExampleOne_returnCorrectOutput() {
        String expected = "1[.]1[.]1[.]1";
        String actual = testee.defangIPaddr(
                "1.1.1.1"
        );
        assertEquals(expected, actual);
    }

    @Test
    public void defangIPaddr_inputExampleTwo_returnCorrectOutput() {
        String expected = "255[.]100[.]50[.]0";
        String actual = testee.defangIPaddr(
                "255.100.50.0"
        );
        assertEquals(expected, actual);
    }

}

File: android/kotlin/src/test/kotlin/com/deanalvero/kotlin/answers/leetcode/easy/DefangingAnIPAddressTest.kt
package com.deanalvero.kotlin.answers.leetcode.easy

import org.junit.Assert
import org.junit.Before
import org.junit.Test

class DefangingAnIPAddressTest {

    lateinit var testee: DefangingAnIPAddress

    @Before
    fun setUp() {
        testee = DefangingAnIPAddress()
    }

    @Test
    fun defangIPaddr_inputExampleOne_returnCorrectOutput() {
        val expected = "1[.]1[.]1[.]1"
        val actual = testee.defangIPaddr(
                "1.1.1.1"
        )
        Assert.assertEquals(expected, actual)
    }

    @Test
    fun defangIPaddr_inputExampleTwo_returnCorrectOutput() {
        val expected = "255[.]100[.]50[.]0"
        val actual = testee.defangIPaddr(
                "255.100.50.0"
        )
        Assert.assertEquals(expected, actual)
    }

}